lib/test_queue/runner/rspec.rb

```diff
@@ -11,25 +11,129 @@
 end
 
 module TestQueue
+  class << self
+    # give ExampleGroups 'n such a quick/easy way to get at this worker's
+    # iterator
+    attr_accessor :iterator
+  end
+
   class Runner
     class RSpec < Runner
+      SPLIT_GROUPS = ["1", "true"].include?(ENV.fetch("TEST_QUEUE_SPLIT_GROUPS", "0").downcase)
+
       def initialize
         @rspec = ::RSpec::Core::QueueRunner.new
         super(@rspec.example_groups.sort_by{ |s| -(stats[s.to_s] || 0) })
+
+        if SPLIT_GROUPS
+          @group_queues = {}
+          @all_groups = {}
+          @queue.each do |group|
+            group.descendants.each do |subgroup|
+              @group_queues[subgroup.to_s] = ::RSpec.world.filtered_examples[subgroup]
+              @all_groups[subgroup.to_s] = subgroup
+            end
+          end
+        end
+      end
+
+      def has_own_examples?(group)
```

Comment:
I'm having trouble with this terminology. Maybe "has_ descendant_examples_in_queue?" and "has_examples_in_queue?"